Framework/Algorithms/src/GetEi.cpp

```diff
@@ -325,10 +325,17 @@ void GetEi::extractSpec(int64_t wsInd, double start, double end) {
 
   childAlg->setProperty<MatrixWorkspace_sptr>("InputWorkspace",
                                               getProperty("InputWorkspace"));
+
   childAlg->setProperty("XMin", start);
   childAlg->setProperty("XMax", end);
-  childAlg->setProperty("StartWorkspaceIndex", wsInd);
-  childAlg->setProperty("EndWorkspaceIndex", wsInd);
+  if (wsInd < std::numeric_limits<int>::max()) {
```

Comment:
I don't understand why this function's signature takes an `int64_t` when the child algorithm needs and `int`. It looks like version 2 has the same issue but hides it with a `static_cast<int>`. 

Would changing `GetEi::extractSpec` so that `wsInd` is an `int` or changing `CropWorkspace` so that `StartWorkspaceIndex` and `EndWorkspaceIndex` are `int64_t` be better solutions? The former advertises a limitation, while the later may introduce other bugs but makes it work as a naive developer would assume.